### PR TITLE
Remove SYSTEM_FILE check in test_system_file_never_expired

### LINKED ISSUE #9497: test_system_file_never_expired failing when building wheels
The ``test_system_file_never_expired`` test is causing failures when building our wheels:

https://travis-ci.org/MacPython/astropy-wheels/builds/603556155?utm_medium=notification&utm_source=github_status

The test failure is:

```
______________ TestDefaultAutoOpen.test_system_file_never_expired ______________
self = <astropy.utils.iers.tests.test_leap_second.TestDefaultAutoOpen object at 0x7f31c09c2908>
tmpdir = local('/tmp/pytest-of-root/pytest-0/test_system_file_never_expired0')
    @pytest.mark.skipif(not os.path.isfile(SYSTEM_FILE),
                        reason=f'system does not have {SYSTEM_FILE}')
    def test_system_file_never_expired(self, tmpdir):
        self.remove_auto_open_files('erfa')
        with iers.conf.set_temp('system_leap_second_file', SYSTEM_FILE):
            ls = iers.LeapSeconds.open()
            assert ls.expires > self.good_enough
            assert ls.meta['data_url'] in (iers.IERS_LEAP_SECOND_FILE,
                                           SYSTEM_FILE)
    
            # Also check with a "built-in" file that is expired
            fake_file = make_fake_file('28 June 2017', tmpdir)
            iers.LeapSeconds._auto_open_files[0] = fake_file
            ls2 = iers.LeapSeconds.open()
            assert ls2.expires > Time.now()
>           assert ls2.meta['data_url'] == SYSTEM_FILE
E           AssertionError: assert 'https://hpie...ap_Second.dat' == '/usr/share/z...-seconds.list'
E             - https://hpiers.obspm.fr/iers/bul/bulc/Leap_Second.dat
E             + /usr/share/zoneinfo/leap-seconds.list
/venv/lib/python3.7/site-packages/astropy/utils/iers/tests/test_leap_second.py:241: AssertionError
=============================== warnings summary ===============================
```

This seems to happen only for the Linux builds.

@mhvk - any idea what could be causing this?

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix #9497

### CODE REVIEW

**Comment by mhvk** [comment]
Hmm, sorry to have forgotten about this. I think there are 3 possible solutions (all fine by me, but a bit different from what you have here!):
1. Just remove the test altogether - as is, all it does is to warn people if their system is out of date (and us if travis is out of date).
2. Skip the test on machines where we build wheels.
3. Ensure the machines where we build wheels are up to date.

(3) was easy on travis, but I'm not sure about the build machines; also not sure how to implement (2), i.e., detect the build machines. (1) is easiest but seems a bit sad (but can also just be an open issue for now).

@astrofrog - any suggestions?

**Comment by bsipocz** [comment]
I only do sdist locally, we build all the wheels on travis/appveyor where, I believe, there should be no previous version of the file around.

**Comment by mhvk** [comment]
Oops, @bsipocz, didn't realize you were in charge of that as well these days! So, would it be possible on travis to add the same change that I made for testing, see https://github.com/astropy/astropy/pull/9365/commits/7bfc65a461f917f3388fb2e7a250da80340dab7f (for `appveyor`, it doesn't matter, it doesn't have the file anyway).

**Comment by bsipocz** [comment]
Yeah, much of that stuff ends up with releasing. The actual wheel maintenance mostly falls on @astrofrog these days, but that still doesn't change the fact that we use CI infrastructures for the building part rather than local systems.

**Comment by bsipocz** [comment]
tzdata has been added to the linux wheels dockers, so I would say go ahead and remove the test as if the only purpose is to notify people about the system it shouldn't really live in the test suite anyway. 🤷‍♀

**Comment by mhvk** [comment]
Hmm, the test *does* serve the purpose of testing that if such a system file is available, it is actually used! Ideally, we find out where similar files live in other systems to set us up for auto-updating without needing a download (well, ideally erfa itself does that at some point...).

**Comment by bsipocz** [comment]
Can this, or any of the other proposals go ahead?

**Comment by pllim** [comment]
Any resolution for this so that the release can move forward?

**Comment by pllim** [comment]
> we find out where similar files live in other systems to set us up for auto-updating without needing a download

@mhvk , how do we accomplish that?

**Comment by olebole** [comment]
I just found that on Debian, I now also [see this failure in our CI tests](https://ci.debian.net/data/autopkgtest/testing/amd64/a/astropy/4179764/log.gz) -- since a few days (Feb 1?). I still have to investigate the circumstances a bit more, however.

**Comment by pllim** [comment]
@olebole , you might want to look at #9908, #9910, and #9912 . There are a series of somewhat related patches and might fix your error.

@mhvk , if you think this is too drastic and not needed anymore, please close it without merge.

Thank you!

**Comment by mhvk** [comment]
@pllim - I'll await what @olebole finds. I fear I've given us more work than needed with this test, but I'll give it one more chance before just removing it!

**Comment by pllim** [comment]
@olebole , is this patch still needed? Please advise. Thanks!

**Comment by bsipocz** [comment]
We still have to patch the wheels, so doing something about it would be nice.

https://github.com/MacPython/astropy-wheels/blob/master/patches/test_system_file_never_expired.patch

**Comment by mhvk** [comment]
Maybe I should simply rewrite the test so that it always tests something like 2018 rather than the current date - that way we still check that `SYSTEM_FILE` can be used but stop relying on it being up to date, which is really not astropy's problem.

**Comment by mhvk** [comment]
I'm confused - I actually think I already fixed this in #9908 - is this still needed at all?

**Comment by bsipocz** [comment]
I've tried to remove the patch when built the wheels for 4.0.1, but unfortunately it didn't work:

https://travis-ci.org/github/MacPython/astropy-wheels/builds/667579720?utm_source=github_status&utm_medium=notification

(note that there is another fits related failure, but the file expiration one is also present for the linux builds)

**Comment by mhvk** [comment]
But I think this particular work-around is out-of-date. The test that failed is skipped on CI:
https://github.com/astropy/astropy/blob/7e6e4b90fc37c8df4a824833bb934b04a5789d12/astropy/utils/iers/tests/test_leap_second.py#L16

Might we be able to add something similar for wheel building?

**Comment by pllim** [comment]
I updated the PR based on https://github.com/astropy/astropy/issues/10403#issuecomment-634709067 . Is this what you meant, @mhvk ?

File: astropy/utils/iers/tests/test_leap_second.py [hunk 1]
```diff
@@ -80,7 +79,7 @@ def test_open_leap_seconds_list(self, file):
     def test_open_system_file(self):
         ls = iers.LeapSeconds.open(SYSTEM_FILE)
         expired = ls.expires < Time.now()
-        if ON_CI and expired:
+        if expired:
             pytest.skip("CI system leap second file is expired.")
```
Comment (mhvk) [inline astropy/utils/iers/tests/test_leap_second.py:83]: Delete "CI" in the message too, I guess (below as well)

Comment (pllim) [reply astropy/utils/iers/tests/test_leap_second.py]: Done. Thanks for the review!

**Pull request merged**